src/functions-and-arrays.js

```diff
@@ -1,41 +1,114 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(a, b) {
+
+  let result = 0
+
+  if (a <= b) {
+
+    result = b
+
+  } else {
+
+    result = a
+
+  }
+  return result
+}
+
 
 
 
 // Iteration #2: Find longest word
+
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+
+function findLongestWord(words) {
+
+  if (words.length === 0) {
+    return null
+  }
+
+  let longuestWord = ''
+
+  for (let i = 0; i < words.length; i++) {
+
+    if (longuestWord.length < word[i].length) {
+
+      longuestWord = word[i]
+    }
+
+  }
+  return longuestWord
+}
```

Comment:
- En el `if` de la línea 36, comparas el largo de la variable `longestWord` (`longestWord.length`) con el largo de la palabra sobre la que está iterando, pero has puesto `word`, en vez de `words`. No tienes ninguna variable que se llame `words` (entiendo que lo estás intentando hacer es acceder a la posición `i` del array `words` y una vez accedido a ese valor, saber su largo (`length`). Para ello el código correcto sería `word[i].length`.
- Lo mismo te pasa en la línea 38, ya que asignas a la variable `longestWord`, `word[i]`, pero la variable `word` no existe, es `words`, que es el array.

```suggestion
function findLongestWord(words) {

  if (words.length === 0) {
    return null
  }

  let longuestWord = ''

  for (let i = 0; i < words.length; i++) {

    if (longuestWord.length < words[i].length) {

      longuestWord = words[i]
    }

  }
  return longuestWord
}
```